### PR TITLE
Nitiwiki next

### PR DESCRIPTION
**Pull request by Morriar**
Some fixes for nitiwiki in prevision for the nitlanguage.org site migration and the resolution of #824.

Do not consider b481cea since it belongs to #1368 (and break the loader)

Feature summary:
* ini accept array notation
* customizable sidebar content
* customizable sidebar position
* wikilinks support external links
* better search by title
* better search by path
* make breadcrumbs and summaries optionnal
* add link to github for last_changes and edit mode

### CODE REVIEW

File: contrib/nitiwiki/src/wiki_links.nit [hunk 1]
```diff
@@ -118,6 +120,19 @@ redef class WikiEntry
 		end
 		return null
 	end
+
+	private var md_proc: NitiwikiMdProcessor is lazy do
+		return new NitiwikiMdProcessor(wiki, self)
+	end
+
+	# Process iwikilinks from sidebar.
```
Comment (R4PaSs) [inline contrib/nitiwiki/src/wiki_links.nit]: `iwikilinks` ? Typo or did I miss something ?

**Comment by R4PaSs** [comment]
:shipit: !

File: contrib/nitiwiki/src/wiki_links.nit (hunk 1 above)
Comment (Morriar) [inline contrib/nitiwiki/src/wiki_links.nit]: It's a typo, should be `wikilinks`

**Comment by Morriar** [comment]
Reroll with @R4PaSs remarks

File: contrib/nitiwiki/src/wiki_base.nit [hunk 2]
```diff
@@ -209,6 +209,26 @@ class Nitiwiki
...
+	private fun load_sideblock(name: String): nullable String do
+		if not has_sideblock(name) then
+			message("Error: can't load sideblock `{name}`", 0)
+			return null
+		end
+		name = "{name}.{config.md_ext}"
+		var path = expand_path(config.root_dir, config.sidebar_dir, name)
+		var file = new FileReader.open(path)
+		var res = file.read_all
+		file.close
```
Comment (privat) [inline contrib/nitiwiki/src/wiki_base.nit]: there is a oneliner `path.to_path.read_all` that does the opening/closing if I remember correclty

File: contrib/nitiwiki/src/wiki_base.nit [hunk 3]
```diff
@@ -560,6 +583,28 @@ class WikiArticle
 	redef fun to_s do return "{name} ({parent or else "null"})"
 end
 
+# The sidebar is displayed in front of the main panel of a `WikiEntry`.
+class WikiSidebar
+
+	# Wiki used to parse sidebar blocks.
+	var wiki: Nitiwiki is lazy do return entry.wiki
```
Comment (privat) [inline contrib/nitiwiki/src/wiki_base.nit]: what is the point of a `var` here? a `fun` would be enough as there is no real need to store more state than the related entry.

File: contrib/nitiwiki/src/wiki_base.nit [hunk 4]
```diff
@@ -702,6 +747,59 @@ class WikiConfig
...
+	# Position of the sidebar between `left`, `right` and `none`. Any other value
+	# will be considered as `none`.
+	#
+	# * key: `wiki.sidebar`
+	# * default: `left`
+	var sidebar: String is lazy do
+		return value_or_default("wiki.sidebar", "left")
+	end
+
+	# Sidebar markdown block to include.
```
Comment (privat) [inline contrib/nitiwiki/src/wiki_base.nit]: what is a block?

**Comment by privat** [comment]
This is nice. Have you considered fixing #1368 ?

**Comment by Morriar** [comment]
Reroll rebased on #1368

**Pull request merged**